src/main/java/com/cloudbees/plugins/deployer/sources/MavenArtifactDeploySource.java

```diff
@@ -158,7 +158,7 @@ public String getGroupId() {
      */
     @NonNull
     public String getType() {
-        return type == null ? DEFAULT_TYPE : type;
+        return type;
```

Comment:
🐛 I am pretty sure it was a migration logic. Should be replaced by `readResolve`